File: pinyougou/pinyougou-cart/pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/AddressService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojo.TbAddress;
import com.pinyougou.pojo.TbAreas;
import com.pinyougou.pojo.TbCities;
import com.pinyougou.pojo.TbProvinces;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;

import java.util.List;
import java.util.Map;

public interface AddressService extends BaseService<TbAddress> {

    PageResult search(Integer page, Integer rows, TbAddress address);

    List getAddressInformation(String username);

    //传入一个TbAddress一个map，就会自动给你将id转成地名
    Map<String, Object> addressIdToName(TbAddress tbAddress, Map<String, Object> addressInformation);

    void setDefaultAddress(String addressId, String username);

    void updateByPrimaryKey(TbAddress tbAddress);

    List<TbProvinces> findProvince();

    List<TbCities> findCities(String provinceId);

    List<TbAreas> findAreas(String cityId);

    void insertAddressInformation(TbAddress tbAddress);
}

File: pinyougou/pinyougou-user-web/src/main/webapp/js/service/addressService.js
app.service("addressService",function($http){
    this.getAddressInformation = function () {
        return $http.get("address/getAddressInformation.do?r=" + Math.random());
    };
    this.setDefaultAddress = function (addressId) {
        return $http.get("address/setDefaultAddress.do?addressId=" + addressId);
    };
    this.delete = function (selectedIds) {
        return $http.get("address/delete.do?ids=" + selectedIds);
    };
    this.findOne = function (id) {
        return $http.get("address/findOne.do?id=" + id +"&r="+ Math.random());
    };
    this.updateAddressInfromation = function (thisUserInfromation) {
        return $http.post("address/updateAddressInfromation.do?r="+ Math.random(),thisUserInfromation)
    };
    this.insertAddressInformation = function (thisUserInfromation) {
        return $http.post("address/insertAddressInformation.do?r="+ Math.random(),thisUserInfromation)
    };


    this.findProvince = function () {
        return $http.get("address/findProvince.do?r=" + Math.random())
    };
    this.findCities = function (provinceId) {
        return $http.get("address/findCities.do?provinceId=" + provinceId +"&r="+ Math.random())
    };
    this.findAreas = function (cityId) {
        return $http.get("address/findAreas.do?cityId=" + cityId +"&r="+ Math.random())
    };
});